src/mlpack/core/data/tokenizer/strtok.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file strtok.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of Strtok class which tokenizes using single character.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STRTOK_HPP
+#define MLPACK_CORE_DATA_STRTOK_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+
+namespace mlpack {
+namespace data {
+/**
+ * A simple Strtok class
+ */
+class Strtok
+{
+ public:
+  /**
+  * A constructor to set deimiter.
+  * @param Delimiter A string which you want to use as delimiter
+  */
+  Strtok(std::string Delimiter)
```

Comment:
It might be easier to use initialization lists here: `Strtok(const std::string& delimiter) : delimiter(delimiter) { }`.